app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsController.scala

```diff
@@ -55,12 +64,33 @@ class GroupMemberGrsController @Inject() (
     (authenticate andThen journeyAction).async {
       implicit request =>
         saveRegistrationDetails(journeyId).flatMap {
-          case Right(_) =>
-            Future.successful(Redirect(routes.OrganisationListController.displayPage()))
+          case Right(registration) =>
+            registrationStatus(registration).map {
+              case STATUS_OK => Redirect(routes.OrganisationListController.displayPage())
+              case DUPLICATE_SUBSCRIPTION =>
+                Redirect(pptRoutes.NotableErrorController.duplicateRegistration())
+            }
           case Left(error) => throw error
         }
     }
 
+  private def registrationStatus(
+    registration: Registration
+  )(implicit hc: HeaderCarrier): Future[RegistrationStatus] = {
+    val organisationDetails: Option[OrganisationDetails] =
+      registration.groupDetail.map(_.members.last).flatMap(_.organisationDetails)
+    val businessPartnerId: String = organisationDetails.flatMap(_.businessPartnerId).get
```

Comment:
Does this need protection against None?   I think we'll need to fail more gracefully if businessParnerId not returned from GRS.

Reply:
I agree with @SteveSugden. Can we just bypass the status check if we don't have a business partner id?

Reply:
Done